### PR TITLE
Add /debug/mesh endpoint

### PR DESCRIPTION
**Pull request by howardjohn**
Currently no way to see current mesh config

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/xds/debug.go [hunk 1]
```diff
@@ -625,6 +626,13 @@ func (s *DiscoveryServer) InjectTemplateHandler(webhook *inject.Webhook) func(ht
 	}
 }
 
+// MeshHandler dumps the mesh config
+func (s *DiscoveryServer) MeshHandler(w http.ResponseWriter, r *http.Request) {
+	if err := (&jsonpb.Marshaler{}).Marshal(w, s.Env.Mesh()); err != nil {
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/debug.go:631]: I think we need indents here

Comment (howardjohn) [reply pilot/pkg/xds/debug.go:631]: done

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**